File: zig/isogram/isogram.zig
pub fn isIsogram(str: []const u8) bool {
    var letter_array = [_]u8{0} ** 26;
    for (str) |ch| {
        const ch_h = ch | 32;
        if (ch_h >= 'a' and ch <= 'z') {
            const array_index: u8 = ch_h - 'a';
            if (letter_array[array_index] > 0) {
                return false;
            }
            letter_array[array_index] += 1;
        }
    }
    return true;
}

File: c/isogram/isogram.c
#include "isogram.h"

bool is_isogram(const char phrase[]) {
    if (phrase == NULL) {
        return false;
    }
    int letter_array[26] = {0};
    int i = 0;
    char ch = phrase[0];
    while (ch != '\0') {
        printf("%c, ", ch);
        if (isalpha(ch)) {
            int array_index = tolower(ch) - 'a';
            printf("%d, letter_array %d \n", array_index, letter_array[array_index]);
            if (letter_array[array_index]) {
                return false;
            }
            ++letter_array[array_index];
        }
        ch = phrase[++i];
    }
    return true;
}

int main(void) {
  int result = is_isogram("cffrefg");
  printf("%d \n", result);

  return 0;
}
